clippy_lints/src/methods/mod.rs

```diff
@@ -107,6 +108,29 @@ declare_clippy_lint! {
     "used `cloned` where `copied` could be used instead"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for usage of `_.cloned().last()`.
+    ///
+    /// ### Why is this bad?
```

Comment:
Not only performance, but also style: For `.cloned().next()`, the call will very likely be inlined anyway, but as code changes, keeping a consistent style is also a good thing.

I'd also add a `### Known Problems` section, because the lint is partially behavior-changing, because `.cloned().last()` will clone all elements unlike `.last().cloned()`. This cloning could be observed as a side effect, which would be removed by this lint. Given the contract of `Clone` and the likely perf benefit, I find this acceptable, but we should still document it.